docs/architecture/adr-020-protobuf-transaction-encoding.md

```diff
@@ -135,40 +166,155 @@ Then, each module's client handler will at the minimum accept a `Marshaler` inst
 of a concrete Amino codec and a `Generator` along with an `AccountRetriever` so
 that account fields can be retrieved for signing.
 
-#### Interface `oneof` Handling
+### Interface `oneof` Handling
+
+If a module needs to work with `sdk.Msg`s that use interface types, those interface
+types should be dealt with using `google.protobuf.Any` for signing and a `oneof`
+at the app-level for encoding.
 
-If the module needs to work with any `sdk.Msg`s that use interface types, that
-`sdk.Msg` should be implemented as an interface with getters and setters on the
-module level and a no-arg constructor function should be passed around to
-required CLI and REST client commands.
+Using `google.protobuf.Any` for signing will allow client libraries to create
+reusable code for dealing with interface types that doesn't require regenerating
+protobuf client libraries for every chain. Using `oneof`s at the encoding level
```

Comment:
Is "interface" even a defined term in protobuf? Why would the field numbers (in .proto) be the same for every implementation of an interface?

Reply:
Interface refers to go interfaces. You may need to look at some of the existing encoding approach and the migration from amino to understand the context here